2023/adv2023-rust/src/day11.rs

```diff
@@ -1,3 +1,115 @@
+use std::collections::{HashMap, HashSet};
+use std::io::{self, BufRead};
+// TODO: ^ Why BufRead if it's not used? What's self, just the io namespace?
+
 fn main() {
-    println!("Hello, world!");
+    println!("Parse image");
+    let img = parse_input();
+    println!("  {} x {}", img.len(), img[0].len());
+
+    // Expand image
+    println!("Expand image");
+    let img = expand_img(img);
+    println!("  {} x {}", img.len(), img[0].len());
+
+    // Find coords of galaxies
+    println!("Find coords of galaxies");
+    let mut galaxies = Vec::new();
+
+    for i in 0..img.len() {
+        for j in 0..img[0].len() {
+            if img[i][j] == '#' {
+                galaxies.push((i, j));
+            }
+        }
+    }
+
+    // Find distances between each pair of galaxies
+    println!("Find distances between each pair of galaxies");
+    let mut distances: HashMap<String, usize> = HashMap::new();
+    for x in 0..galaxies.len() {
+        for y in x+1..galaxies.len() {
+            let src = galaxies[x];
+            let dest = galaxies[y];
+            let key = get_pair_key_str(&src, &dest);
+
+            // Manhattan distance
+            let dist = abs_diff(src.0, dest.0) + abs_diff(src.1, dest.1);
```

Comment:
`(src.0 - dest.0).abs()`?

Reply:
They're `usize`s, so I think I have to cast to something signed to do that